wordpress-mobile/WordPress-iOS
# Stats Revamp: Appearance Update

Refs #18236. This PR fixes up some remaining stats revamp appearance issues:

* Adds titles to error cells and removes top and bottom borders
* Adds titles to some remaining no data rows
* Simplifies the way the Add New Stats Card row was created to decouple it from the other cell implementations

This PR has the side effect of also removing top and bottom borders from cells in the Period view when the revamp feature flags are enabled. It would have required quite a bit of reworking to keep them enabled there, and I think visually it looks fine. Also for consistency, new sections like the dashboard have no kind of cell borders, and neither does the Reader.

Error cells and Publicise and Tags & Categories No Data states

| Before | After |
|---|---|
| ![Simulator Screen Shot - iPhone 13 Pro - 2022-06-16 at 15 10 38](https://user-images.githubusercontent.com/4780/174617064-eb0c3e81-a96a-4442-907b-26ae9631d0b3.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-06-20 at 14 31 31](https://user-images.githubusercontent.com/4780/174616916-8ccc4551-d2e9-4ea1-b8b4-0be0b86ed0e8.png) |

Loading states in Insights and Period views
| Insights | Period |
|---|---|
| ![Simulator Screen Shot - iPhone 13 Pro - 2022-06-20 at 14 54 41](https://user-images.githubusercontent.com/4780/174617363-b9248853-5396-49a4-b3f5-33fdcd0735bf.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-06-20 at 14 30 23](https://user-images.githubusercontent.com/4780/174617019-d42e8997-0088-4996-942e-4c0d8672671b.png) |

**To test**

* Build and run with the feature flags enabled
* View Insights and Period views for a range of sites and ensure cells look as they currently do on production. In particularly check loading and error / no data states if possible.

* Enable the revamp feature flags and re-check the above
* I found I had to use Network Link Conditioner to slow my connection down to be able to see the loading states for long enough in Insights, but your mileage may vary

## Regression Notes

1. Potential unintended areas of impact
None other than mentioned above.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
